blog/2024-07-07-openhab-4-2-release.md

```diff
@@ -0,0 +1,276 @@
+[@mueller-ma](https://github.com/mueller-ma) and Danny Baumann ([@maniac103](https://github.com/maniac103)) for all their work on the apps!
+
+## Core Runtime Enhancements
+
+### Persistence Extensions
+
+_Mark Herwege ([@mherwege](https://github.com/mherwege)), openHAB Contributor_
+
+Persistence Extensions have allowed the different scripting languages to easily persist states and retrieve persisted states or values calculated from these persisted states for a long time.
+
+As openHAB supports units of measurument (UoM), persistence services can also return persisted values with a unit of measurement.
+However, the Persistence Extensions did not consider these units of measurement and returned just a decimal value as the result of their calculations.
+With openHAB 4.2, Persistence Extension have been enhanced to also return the unit of measurement.
+This allows rules and scripts to simply calculate with `QuantityType`s without any implicit or explicit conversion to a `DecimalType`, while keeping the results consistent if different units of measure are used.
+Note that this change may require adapting previous rules.
+
+With the introduction of [Time Series support](/blog/2023-12-22-openhab-4-1-release.html#time-series-support-for-forecasts-and-historical-values) in openHAB 4.1, persistence services were extended to allow storing future values and forecasts.
+openHAB 4.2 extends the Persistence Extensions to allow persisting future states and retrieving future values.
+All persistence data retrieval methods now have `last`, `next` and `between` variants, 
+whereby `last` will use the period from a point in time in the past until now, `next` will use the period from now to a future point in time and `between` will use the period between two provided points in time in the past and/or future.
+
+To allow storing multiple values to persistence in one go, Persistence Extensions methods now support working with Time Series instead of individual values when persisting or retrieving data.
+Time Series are series of time/value pairs, and can be constructed in rules, e.g. by modifying a Time Series retrieved from another persistence service or another Item.
+
+Also, there was no way before to retrieve the time of the last state change for an item from persistence.
+The existing methods (`previousState()` or `previousState(true)`) would not give the right result if the same value had been persisted multiple times.
+A new method `lastChange()` has been added to resolve this.
+
+Scripting languages (Rules DSL, JavaScript, jRuby and Blockly) have implemented full support of these new and enhanced Persistence Extensions, please refer to the individual languages' documentation for more information.
+
+As much has changed, it is worthwhile to read the [documentation](/docs/configuration/persistence.html#persistence-extensions-in-scripts-and-rules) to update your knowledge about openHAB persistence extensions and learn everything about the new features.
+
```

Comment:
A few words about Java21, hopefully not too technical....
Please feel free to change the test....

```suggestion

### Java 21

_Holger Friedrich ([@holgerfriedrich](https://github.com/holgerfriedrich)), openHAB maintainer_

Java 17 is required for all releases since openHAB 4.0, and Java 17 is still recommended for running openHAB.
But with version 4.2, we introduce Java 21 as an alternative.

We typically focus on Java releases with long-term support (LTS).
The next LTS release following after Java 17 is Java 21, released in 2023.
To allow for compiling openHAB with Java 21, we had to adapt our toolchain, upgrade dependencies, and also introduce some code changes.
This resulted in a comprehensive upgrade of almost all dependencies used in core. Just to name the most important ones: Karaf, Xtext/Xtend.
To allow for running openHAB with Java 21, a few more changes had to be introduced:
One notable difference between Java 17 and Java 21 is the change of the serialization format of DateTime in CLDR 42, which required some extra efforts in persistence to allow switching the Java version in an existing installation.

You can use either Java 17 or Java 21 to run your openHAB instance.
The openHAB distribution uses the Java 17 binary format which can run with both Java versions.
Please consider Java 21 as experimental, as it has not been tested thoroughly yet.
Note that Java 21 is only available for 64-bit systems.
If you use openHABian, you can install Java 21 using `openhabian-config`.
Please note that this is only possible if you installed the native 64-bit image.

Developers can even choose between three options: Java 17, Java 21 creating Java 17 binaries (default for OH if you use Java 21), and native Java 21 (binaries do not work on a Java 17 server).
Whereas we recommend Java 17 for running openHAB, development can be safely done on Java 21.
See the developer documentation for more information.
```

Reply:
Thanks, I have integrated this and just pushed to commit.